Feature: Add return value to wp_delete_file() function

Debarghya-Banerjee opened the pull request:
**Trac Ticket - [Core-61590](https://core.trac.wordpress.org/ticket/61590)**

## **Summary**

This Pull Request addresses the issue where using `wp_delete_file()` lacks a return value, making it challenging to verify the success of file deletion operations compared to `unlink()`.

## **Description**

### **Problem**

The `wp_delete_file()` lacks a return value, making it challenging to verify the success of file deletion operations without doing overhead checks if the file still exists.

### **Example**

```
wp_delete_file( $file ) => returns void
```

### **Solution**

Since the `wp_delete_file()` internally uses the `unlink` function which returns a `boolean` value based on success or failure, to resolve this issue, the return value of `unlink()` is returned i.e., a boolean value is return to make the user aware if the deletion was successful or not. Additionally, if the file is not found, it returns false.

Commit:
Add return value to wp_delete_file() function

mukeshpanchal27 commented on src/wp-includes/functions.php:
```diff
@@ -7646,6 +7646,7 @@ function wp_validate_boolean( $value ) {
  * @since 4.2.0
  *
  * @param string $file The path to the file to delete.
+ * @return bool Returns `true` if the file has been deleted successfully or `false` on failure.
```
```suggestion
 * @return bool True if the file has been deleted successfully, false on failure.
```

Debarghya-Banerjee committed:
Update the Doc comments

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

ellatrix committed:
Section Styles: Prevent flash of variation style updates.

This is fixed by preloading the styles.

Fixes #61589.
Fixes https://github.com/WordPress/wordpress-develop/pull/6989.
See https://github.com/WordPress/gutenberg/pull/63172.

Props aaronrobertshaw, andrewserong, ramonopoly.



git-svn-id: https://develop.svn.wordpress.org/trunk@58690 602fd350-edb4-49c9-b593-d223f7449a82

ellatrix committed:
Section Styles: Fix ref values within block style variations.

Fixes #61589.
Fixes https://github.com/WordPress/wordpress-develop/pull/6989.
See https://github.com/WordPress/gutenberg/pull/63172.

Props aaronrobertshaw, andrewserong, ramonopoly.



git-svn-id: https://develop.svn.wordpress.org/trunk@58691 602fd350-edb4-49c9-b593-d223f7449a82

ellatrix committed:
Editor: Update packages for 6.6 RC 3.

Fixes #61603.
Fixes https://github.com/WordPress/wordpress-develop/pull/6998.
See https://make.wordpress.org/core/handbook/about/release-cycle/block-editor-release-process-for-major-releases/#package-updates-and-core-patches.

Props ellatrix, youknowriad.



git-svn-id: https://develop.svn.wordpress.org/trunk@58693 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
REST API: Remove a few unused variables in `foreach` loops.

Follow-up to [38832], [48173], [49132], [49925].

Props antonvlasenko.
See #61593.

git-svn-id: https://develop.svn.wordpress.org/trunk@58694 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
HTML API: Correct node name in `generate_implied_end_tags()`.

The `generate_implied_end_tags()` algorithm has been comparing the
current node to a list of node names, which means that it won't ever
pop any elements from the stack of open elements.

This patch corrects the mistake by comparing node name against the
list, thus fixing the algorithm. This was noted in development work
for the 6.7 release.

Developed in https://github.com/WordPress/wordpress-develop/pull/6988
Discussed in https://core.trac.wordpress.org/ticket/61576

Props dmsnell, jonsurrell.
See #61576.


git-svn-id: https://develop.svn.wordpress.org/trunk@58702 602fd350-edb4-49c9-b593-d223f7449a82

tellthemachines committed:
Editor: enqueue block custom CSS only when block renders on the page.

Updates the global styles custom CSS handling logic to be consistent with other global styles and take advantage of conditional enqueuing of block styles.

Props isabel_brison, aaronrobertshaw, andrewserong.
Fixes #61395.


git-svn-id: https://develop.svn.wordpress.org/trunk@58703 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
Docs: Correct `@return` values for a few REST API class methods.

Includes listing the expected type first, instead of `WP_Error`.

Follow-up to [39031], [39033], [46696], [49927], [49929], [50993], [51286], [51973], [52079], [52286], [53152], [56415].

Props antonvlasenko.
See #61593.

git-svn-id: https://develop.svn.wordpress.org/trunk@58704 602fd350-edb4-49c9-b593-d223f7449a82

hellofromtonya committed:
REST API: Ensure string returned in WP_REST_Templates_Controller::get_wp_templates_author_text_field().

Adds a fail-safe to return an empty string should the `switch` ever fall through without returning.

Currently, `WP_REST_Templates_Controller::get_wp_templates_author_text_field()` is tightly coupled to `WP_REST_Templates_Controller::get_wp_templates_original_source_field()`. However, if the `$original_source` values change in either method, but not both, it is possible a `void` or `null` will be returned, rather than a `string`.

Follow-up to [57366].

Props antonvlasenko, hellofromTonya, debarghyabanerjee.
Fixes #61580.

git-svn-id: https://develop.svn.wordpress.org/trunk@58705 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
Docs: Fix typos in various REST API DocBlocks and comments.

Follow-up to [34928], [43739], [46422], [50717], [51973], [54528].

Props antonvlasenko.
Fixes #61593.

git-svn-id: https://develop.svn.wordpress.org/trunk@58706 602fd350-edb4-49c9-b593-d223f7449a82

hellofromtonya committed:
Docs: Fix multi-line inline comments in WP_REST_Templates_Controller.

Converts single inline comment formatting into multi-line inline comment formatting, per the coding standards.

Ref: https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-inline-comments

Follow-up to [58303], [57366].

See #61608.

git-svn-id: https://develop.svn.wordpress.org/trunk@58707 602fd350-edb4-49c9-b593-d223f7449a82

tellthemachines committed:
Editor: remove Group inner container from Grid variation in classic themes.

Updates the logic in the layout filter that replaces the Group block inner container in classic themes to exclude the Grid variation.

Props isabel_brison, aaronrobertshaw, andrewserong.
See #61635.



git-svn-id: https://develop.svn.wordpress.org/trunk@58708 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
KSES: Add `opacity` to the list of safe CSS properties.

Original PR from Gutenberg repository:
* [https://github.com/WordPress/gutenberg/pull/59891 #59891 Update overlay step function in cover block]

Reference: [https://developer.mozilla.org/en-US/docs/Web/CSS/opacity MDN Web Docs: opacity].

Props sunil25393, wildworks, poena, Mamaduka, presstoke.
Fixes #61536.

git-svn-id: https://develop.svn.wordpress.org/trunk@58709 602fd350-edb4-49c9-b593-d223f7449a82

joemcgill committed:
Editor: Revert caching of global styles for blocks.

This reverts [58334] to fix a bug where edits to block styles made in the site editor were not showing in the front end.

Props joemcgill, spacedmonkey, andrewserong, hellofromtonya, audrasjb.
See #59595.


git-svn-id: https://develop.svn.wordpress.org/trunk@58710 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
HTML API: Join successive text nodes in html5lib test representation.

Many tests from the html5lib test suite fail because of differences in
text handling between a DOM API and the HTML API, even though the
semantics of the parse are equivalent. For example, it's possible in
the HTML API to read multiple successive text nodes when the tokens
between them are ignored.

The test suite didn't account for this and so was failing tests. This
patch improves the construction of the representation to compare
against the test suite so that those tests don't fail inaccurately.

Developed in https://github.com/WordPress/wordpress-develop/pull/6984
Discussed in https://core.trac.wordpress.org/ticket/61576

Props bernhard-reiter, dmsnell, jonsurrell.
See #61576.


git-svn-id: https://develop.svn.wordpress.org/trunk@58712 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
HTML API: Simplify breadcrumb accounting.

Since the HTML Processor started visiting all nodes in a document, both
real and virtual, the breadcrumb accounting became a bit complicated
and it's not entirely clear that it is fully reliable.

In this patch the breadcrumbs are rebuilt separately from the stack of
open elements in order to eliminate the problem of the stateful stack
interactions and the post-hoc event queue.

Breadcrumbs are greatly simplified as a result, and more verifiably
correct, in this construction.

Developed in https://github.com/WordPress/wordpress-develop/pull/6981
Discussed in https://core.trac.wordpress.org/ticket/61576

Follow-up to [58590].

Props bernhard-reiter, dmsnell.
See #61576.


git-svn-id: https://develop.svn.wordpress.org/trunk@58713 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
HTML API: Add context to Unsupported_Exception class for improved debugging.

The HTML Processor internally throws an exception when it reaches HTML
that it knows it cannot process, but this exception is not made
available to calling code. It can be useful to extract more knowledge
about why it gave up, especially for debugging purposes.

In this patch, more context is added to the WP_HTML_Unsupported_Exception
and the last exception is made available to calling code through a new
method, `get_unsupported_exception()`.

Developed in https://github.com/WordPress/wordpress-develop/pull/6985
Discussed in https://core.trac.wordpress.org/ticket/61646

Props bernhard-reiter, dmsnell, jonsurrell.
See #61646.


git-svn-id: https://develop.svn.wordpress.org/trunk@58714 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
Site Health: Correctly display auto-update status for parent theme.

Follow-up to [47835], [48731].

Props apermo.
Fixes #61649. See #61648.

git-svn-id: https://develop.svn.wordpress.org/trunk@58716 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
Docs: Correct documentation for `_wp_preview_meta_filter()`.

The `$single` parameter is passed via the `get_post_metadata` filter and has no default value.

Includes minor code layout fixes for consistency with the formatting of other long conditionals in core.

Follow-up to [56714].

Props rodrigosprimo.
Fixes #61645.

git-svn-id: https://develop.svn.wordpress.org/trunk@58717 602fd350-edb4-49c9-b593-d223f7449a82

dream-encode committed:
Bundled Themes: Bump default theme versions for release with 6.6.

This updates the version of each default theme to the following versions:
- Twenty Ten: 4.2
- Twenty Eleven: 4.7
- Twenty Twelve: 4.3
- Twenty Thirteen: 4.2
- Twenty Fourteen: 4.0
- Twenty Fifteen: 3.8
- Twenty Sixteen: 3.3
- Twenty Seventeen: 3.7
- Twenty Nineteen: 2.9
- Twenty Twenty: 2.7
- Twenty Twenty-One: 2.3
- Twenty Twenty-Two: 1.8
- Twenty Twenty-Three: 1.5
- Twenty Twenty-Four: 1.2

These versions will released in coordination with WordPress 6.6.

Props sabernhardt, shail-mehta, rudlinkon.
Fixes #60701.

git-svn-id: https://develop.svn.wordpress.org/trunk@58718 602fd350-edb4-49c9-b593-d223f7449a82

ellatrix committed:
Editor: Update packages for 6.6 RC 4.

This package update includes only one revert PR:

https://github.com/WordPress/gutenberg/pull/63412

You can confirm the changes in this package update here:

https://github.com/WordPress/gutenberg/commits/wp/6.6/

Fixes #61654.

Props santosguillamot.



git-svn-id: https://develop.svn.wordpress.org/trunk@58719 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Twenty Fifteen: Fixes pullquote issues with text color and border

Pullquote was not showing the color changes. This uses inherit to do that. It also removes the changes from [58368] as this method is preferred.

Props viralsampat, poena, sabernhardt.
Fixes #59801.


git-svn-id: https://develop.svn.wordpress.org/trunk@58724 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Twenty Sixteen: Fixes button block letter spacing control.

The button block had issues with font styling not applying. This not only brings letter-spacing but also line-height, text-transform and font-weight to wp-button-block-buttons. [57300] fixed the separator issue. This fix also added box-shadow none for consistency.

Props darshitrajyaguru97, harshgajipara, shailu25, sabernhardt, alvitazwar052.
Fixes #58609.


git-svn-id: https://develop.svn.wordpress.org/trunk@58725 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Twenty Twenty-One: Fixes alignment of site title and navigation links.

There was a text alignment issue not centering on all screen sizes. This fixes that and centers vertically.

Props vijaysinh9094, poena, sabernhardt.
Fixes #61633.


git-svn-id: https://develop.svn.wordpress.org/trunk@58726 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Twenty Twenty-Two: Fixes search block not showing selected border on bottom.

The border was not showing on search block. This brings in a resolution for that removing the none from border property.

Props viralsampat, poena, sabernhardt.
Fixes #60669.


git-svn-id: https://develop.svn.wordpress.org/trunk@58727 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Twenty Sixteen: Fixes quote block border not changing color.

The border was not changing color when it was changed. This brings that in. It is worth noting the original theme design did not have this but as noted in the ticket this has been in the quote block for about two years so matching the current block styling in this case makes sense.

Props umesh84, desrosj, sabernhardt.
Fixes #56565.


git-svn-id: https://develop.svn.wordpress.org/trunk@58728 602fd350-edb4-49c9-b593-d223f7449a82

ockham committed:
Interactivity API: Use Script Modules filter for store & config data.

A dedicated API exists for passing data to Script Modules implemented in changeset [58579].
Use this Core API instead of a custom implementation for Interactivity API to pass data to the client.

Developed in https://github.com/WordPress/wordpress-develop/pull/6683.

Props jonsurrell, gziolo, luisherranz, cbravobernal.
Fixes #61512.

git-svn-id: https://develop.svn.wordpress.org/trunk@58729 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Twenty Fifteen: Fixes code block family being different in iframe editor.

The font family was not being included for the code block. This adds Inconsolata to the code block within the editor styles.

Props pranitdugad, sabernhardt.
Fixes #61571.


git-svn-id: https://develop.svn.wordpress.org/trunk@58730 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Multiple themes: Verse block font family different on front.

The font family was not showing correctly for the verse block. The approach taken was to match the front end and this involved also removing the styles in [50358].

Props pranitdugad, sabernhardt, poena, pitamdey.
Fixes #61140.


git-svn-id: https://develop.svn.wordpress.org/trunk@58731 602fd350-edb4-49c9-b593-d223f7449a82

karmatosed committed:
Multiple themes: Comments Reply heading and Cancel reply link did not have spacing.

This brings spacing in that was missing for both Twenty Seventeen and also Twenty Nineteen. It was missing between the words.

Props pitamdey, sabernhardt, nidhidhandhukiya, poena.
Fixes #59334.


git-svn-id: https://develop.svn.wordpress.org/trunk@58732 602fd350-edb4-49c9-b593-d223f7449a82

hellofromtonya committed:
HTML API: Fix "${var} in strings" deprecation notice in html5lib test.

Changeset [58712] introduced the following compile time PHP deprecation notice on >= PHP 8.2 test runs:

{{{
Deprecated: Using ${var} in strings is deprecated, use {$var} instead in /var/www/tests/phpunit/tests/html-api/wpHtmlProcessorHtml5lib.php on line 257
PHPUnit 9.6.20 by Sebastian Bergmann and contributors.
}}}

The `${` syntax for string interpolation was deprecated in PHP 8.2 and should not be used anymore.

Ref: https://wiki.php.net/rfc/deprecate_dollar_brace_string_interpolation

Follow-up to [58712].

Props jrf.
See #61530, #59654, #61576.

git-svn-id: https://develop.svn.wordpress.org/trunk@58733 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
HTML API: Add tests confirming comment behavior in HTML Processor.

There was a bug-fix late in the 6.6 cycle in the HTML Processor which
resolved an issue with the wrong name being reported when paused at
processing-instruction look-alike invalid comments, but no tests were
added to enforce the correct behaviors.

This patch adds the missing tests.

Developed in https://github.com/WordPress/wordpress-develop/pull/6765
Discussed in https://core.trac.wordpress.org/ticket/61530

Follow-up to [58304], [58558].

Props dmsnell, jonsurrell.
See #61530.


git-svn-id: https://develop.svn.wordpress.org/trunk@58734 602fd350-edb4-49c9-b593-d223f7449a82

hellofromtonya committed:
Tests: Don't declare nested named function in Tests_Interactivity_API_wpInteractivityAPIFunctions.

Once the `test_process_directives_when_block_is_filtered()` method has run, the named `test_render_block_data()` function declared nested within becomes part of the global namespace, which could cause problems for other tests.

Quite apart from the fact that the name starting with `test_`  is confusing (as methods prefixed with `test_` are supposed to be test methods to be run by PHPUnit).

Using a closure for this callback fixes the issue. Declared as `static` for a micro-optimization.

Follow-up to [57826].

Props jrf, hellofromTonya.
See #61530.

git-svn-id: https://develop.svn.wordpress.org/trunk@58738 602fd350-edb4-49c9-b593-d223f7449a82

hellofromtonya committed:
Tests: Use data provider in Tests_Interactivity_API_wpInteractivityAPIFunctions.

Refactors the following tests to use a data provider with named test cases:
* `test_wp_interactivity_data_wp_context_with_different_arrays()`
* `test_wp_interactivity_data_wp_context_with_different_arrays_and_a_namespace()`
* `test_wp_interactivity_data_wp_context_with_json_flags()`

This is better as:
1. One failing test will not block the other tests from running.
2. Each test is now referenced by name in any error message, making it more straight forward to see which test failed.
3. The test no longer contains multiple assertions.
3. It makes it more straight forward to add additional tests.

Follow-up to [58594], [58234], [57762], [57743], [57742], [57563].

Props jrf.
See #61530.

git-svn-id: https://develop.svn.wordpress.org/trunk@58739 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
HTML API: Remove leading whitespace after removing class names.

In part of a larger review of CSS semantics and behaviors, this patch
takes the opportunity to remove leading whitespace in an updated class
attribute after the first class in the attribute has been removed.

Previously, if the first class name had been removed, the whitespace
that formerly followed it would remain in the class attribute. This
stood in contrast to removing other class names, which removed their
associated whitespace.

There should be no semantic or functional changes in this patch, only
a slightly-large diff for modified HTML documents that looks prettier
when removing the first class name in a class attribute.

Developed in https://github.com/WordPress/wordpress-develop/pull/6933
Discussed in https://core.trac.wordpress.org/ticket/61531

Props dmsnell, jonsurrell.
See #61531.


git-svn-id: https://develop.svn.wordpress.org/trunk@58740 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
HTML API: Test code improvements in virtual node breadcrumb tests.

Follow-up after feedback to newly-introduced tests,
mostly to enhance the message when the tests fail.

Developed in https://github.com/WordPress/wordpress-develop/pull/7030
Discussed in https://core.trac.wordpress.org/ticket/61646

Follow-up to [58592].

Props dmsnell, jonsurrell.
See #61646.


git-svn-id: https://develop.svn.wordpress.org/trunk@58741 602fd350-edb4-49c9-b593-d223f7449a82

dmsnell committed:
Fix phpdoc nullable types in some files.

It was found that in several places in the HTML API and its supporting files,
the wrong form of type annotation was used for optional parameters.

Instead of using `?type`, this patch uses `type|type-of-default-value` instead,
noting where important if the parameter is optional, and if so, what its default
value is.

Developed in https://github.com/WordPress/wordpress-develop/pull/7031
Discussed in https://core.trac.wordpress.org/ticket/61399

Props dmsnell, jonsurrell.
See #61399.


git-svn-id: https://develop.svn.wordpress.org/trunk@58742 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
Docs: Fix typo in a comment in `wp.media.view.Attachment.Details.TwoColumn`.

Follow-up to [41351].

Props devansh2002, mukesh27.
Fixes #61658.

git-svn-id: https://develop.svn.wordpress.org/trunk@58743 602fd350-edb4-49c9-b593-d223f7449a82

dream-encode committed:
Upgrade/Install: Update the `$_old_files` array for 6.6.

Props dd32, audrasjb, hellofromTonya.
Fixes #61665.

git-svn-id: https://develop.svn.wordpress.org/trunk@58744 602fd350-edb4-49c9-b593-d223f7449a82

SergeyBiryukov committed:
Docs: Add missing description for the `display_rows()` method in list table classes.

Follow-up to [15491], [17002], [27301].

Props nikitasolanki1812, narenin, mukesh27, dd32, SergeyBiryukov.
Fixes #61670.

git-svn-id: https://develop.svn.wordpress.org/trunk@58745 602fd350-edb4-49c9-b593-d223f7449a82

joedolson committed:
Editor: Limit scope of resizable menu container CSS.

The CSS to make menu item containers resizable in the admin menu editor was too broadly scoped, and caused classic editor metaboxes to have unconstrained height. Limit the scope of the CSS changes to only impact menu item containers.

Props neotrope, sabernhardt, joedolson.
Fixes #61662.

git-svn-id: https://develop.svn.wordpress.org/trunk@58747 602fd350-edb4-49c9-b593-d223f7449a82

joedolson committed:
Toolbar: Move user and recovery menus to a higher priority.

Following [58215], admin bar items in the `top-secondary` group have a changed visual order. Increase the priority of the user and recovery menu items so nodes added with higher priorities will still be shown visually before the user and recovery menu items, as they were prior to 58215.

The items will appear in the reverse of the previous order, but the new order now matches their priority order, rather than being the opposite.

Props sabernhardt, joemcgill, pbiron, joedolson.
Fixes #61615.

git-svn-id: https://develop.svn.wordpress.org/trunk@58748 602fd350-edb4-49c9-b593-d223f7449a82

Debarghya-Banerjee committed:
Filesystem API: Add a return value for `wp_delete_file()`.

This addresses a discrepancy where using `unlink()` allows for checking if it was successful via the return value, but `wp_delete_file()` did not have a return value, making it impossible to verify the result without doing overhead checks if the file still exists.

This also brings more consistency with the other `wp_delete_*()` functions, specifically:

* `wp_delete_file_from_directory()`
* `wp_delete_post()`
* `wp_delete_post_revision()`
* `wp_delete_attachment()`
* `wp_delete_attachment_files()`
* `wp_delete_comment()`
* `wp_delete_nav_menu()`
* `wp_delete_term()`
* `wp_delete_site()`
* `wp_delete_user()`

Includes adding basic unit tests for `wp_delete_file()`.

Follow-up to [31575].

Props bedas, debarghyabanerjee, mukesh27, SergeyBiryukov.
Fixes #61590.

git-svn-id: https://develop.svn.wordpress.org/trunk@58715 602fd350-edb4-49c9-b593-d223f7449a82

Debarghya-Banerjee commented:
Closing the PR since the issue has already been resolved.
